stonesoup/sensormanager/tests/test_sensormanager.py

```diff
@@ -0,0 +1,167 @@
+    assert dwell_centres[0] != dwell_centres[2]
+    assert dwell_centres[1] != dwell_centres[2]
+
+
+def test_brute_force_choose_actions():
+    time_start = datetime.now()
+
+    tracks = [Track(states=[
+        GaussianState([[1], [1], [1], [1]],
+                      np.diag([1.5, 0.25, 1.5, 0.25] + np.random.normal(0, 5e-4, 4)),
+                      timestamp=time_start),
+        GaussianState([[2], [1.5], [2], [1.5]],
+                      np.diag([3, 0.5, 3, 0.5] + np.random.normal(0, 5e-4, 4)),
+                      timestamp=time_start + timedelta(seconds=1))]),
+              Track(states=[
+                  GaussianState([[-1], [1], [-1], [1]],
+                                np.diag([3, 0.5, 3, 0.5] + np.random.normal(0, 5e-4, 4)),
+                                timestamp=time_start),
+                  GaussianState([[2], [1.5], [2], [1.5]],
+                                np.diag([1.5, 0.25, 1.5, 0.25] + np.random.normal(0, 5e-4, 4)),
+                                timestamp=time_start + timedelta(seconds=1))])]
+
+    transition_model = CombinedLinearGaussianTransitionModel([ConstantVelocity(0.005),
+                                                              ConstantVelocity(0.005)])
+    predictor = KalmanPredictor(transition_model)
+    updater = ExtendedKalmanUpdater(measurement_model=None)
+    reward_function = UncertaintyRewardFunction(predictor, updater)
+
+    all_dwell_centres = []
+
+    for i in range(3):
+        sensorsA = {RadarRotatingBearingRange(
+            position_mapping=(0, 2),
+            noise_covar=np.array([[np.radians(0.5) ** 2, 0],
+                                  [0, 0.75 ** 2]]),
+            position=np.array([[0], [0]]),
+            ndim_state=4,
+            rpm=60,
+            fov_angle=np.radians(30),
+            dwell_centre=StateVector([0.0]),
+            max_range=np.inf,
+        )}
+
+        sensorsB = {RadarRotatingBearingRange(
+            position_mapping=(0, 2),
+            noise_covar=np.array([[np.radians(0.5) ** 2, 0],
+                                  [0, 0.75 ** 2]]),
+            position=np.array([[0], [0]]),
+            ndim_state=4,
+            rpm=60,
+            fov_angle=np.radians(30),
+            dwell_centre=StateVector([0.0]),
+            max_range=np.inf,
+        )}
+
+        sensorsC = {RadarRotatingBearingRange(
+            position_mapping=(0, 2),
+            noise_covar=np.array([[np.radians(0.5) ** 2, 0],
+                                  [0, 0.75 ** 2]]),
+            position=np.array([[0], [0]]),
+            ndim_state=4,
+            rpm=60,
+            fov_angle=np.radians(30),
+            dwell_centre=StateVector([0.0]),
+            max_range=np.inf,
+        )}
+
+        for sensor_set in [sensorsA, sensorsB, sensorsC]:
+            for sensor in sensor_set:
+                sensor.timestamp = time_start
+
+        sensor_managerA = BruteForceSensorManager(sensorsA, reward_function.calculate_reward)
```

Comment:
So the part where we pass `reward_function.calculate_reward` rather than just `reward_function` suggests to me that we don't need to provide `RewardFunction` classes in the codebase. They're just classes which execute a function no? So just provide the function - or just provide an example in a notebook and leave the crafting of the particular function to the experts.

Reply:
We don't necessarily need to provide a reward function class, it's there more as an example of how you can set one up that works with the rest of the sensor management structure and it is convenient to just import the class in notebooks. I could take it out and just include it in the tutorials (PR to follow this one) if that would be better? Or I can include it as just a function but I think the uncertainty reward function works nicely as a class so you don't need to pass a predictor/updater every time step.